src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RulesBuilder.java

```diff
@@ -472,8 +496,16 @@ private Expression normalizeProperty(
         return null;
     }
 
-    private boolean isDuplicatedClass(List<Rule> rules, Filter f) {
-        return rules.stream().anyMatch(r -> r.getFilter().equals(f));
+    private boolean isDuplicatedClass(
+            List<Rule> rules, Filter f, PercentagesManager percMan, int currentIdx) {
+        Optional<Rule> opRule = rules.stream().filter(r -> r.getFilter().equals(f)).findFirst();
+        boolean result = opRule.isPresent();
+        if (percMan != null) {
```

Comment:
Two nested if can be expressed as one if with and-ed conditions

Reply:
fixed